modules/runner/src/main/java/org/apache/ignite/internal/app/IgniteImpl.java

```diff
@@ -806,15 +811,15 @@ public CompletableFuture<Ignite> start(Path configPath) {
                     raftMgr,
                     clusterStateStorage,
                     cmgMgr
-            );
+            ));
 
             clusterSvc.updateMetadata(new NodeMetadata(restComponent.hostName(), restComponent.httpPort(), restComponent.httpsPort()));
 
             restAddressReporter.writeReport(restHttpAddress(), restHttpsAddress());
 
             LOG.info("Components started, joining the cluster");
 
-            return cmgMgr.joinFuture()
+            CompletableFuture<Ignite> startFuture = cmgMgr.joinFuture()
                     //Disable REST component during initialization.
                     .thenAcceptAsync(unused -> restComponent.disable())
```

Comment:
Not related to this PR, but please add `startupExecutor` here, it's clearly missing